src/integrationTest/java/org/opensearch/security/FlsDlsAndFieldMaskingTest.java

```diff
@@ -0,0 +1,1113 @@
+							.dls(String.format("{\"match\":{\"%s\":\"%s\"}}", FIELD_ARTIST, ARTIST_TWINS))
+							.on(FIRST_INDEX_NAME),
+					new TestSecurityConfig.Role("first_artist_reader")
+							.clusterPermissions("cluster_composite_ops_ro")
+							.indexPermissions("read")
+							.dls(String.format("{\"match\":{\"%s\":\"%s\"}}", FIELD_ARTIST, ARTIST_FIRST))
+							.on(SECOND_INDEX_NAME)
+			);
+
+	/**
+	* User who is allowed to see documents on all indices where value of the {@link Song#FIELD_STARS} is less than zero.
+	*/
+	static final TestSecurityConfig.User ALL_INDICES_STARS_LESS_THAN_ZERO_READER = new TestSecurityConfig.User("stars_less_than_zero_reader")
+			.roles(
+					new TestSecurityConfig.Role("stars_less_than_zero_reader")
+							.clusterPermissions("cluster_composite_ops_ro")
+							.indexPermissions("read")
+							.dls(String.format("{\"range\":{\"%s\":{\"lt\":%d}}}", FIELD_STARS, 0))
+							.on("*")
+			);
+
+	@ClassRule
+	public static final LocalCluster cluster = new LocalCluster.Builder()
+			.clusterManager(ClusterManager.THREE_CLUSTER_MANAGERS).anonymousAuth(false)
+			.authc(AUTHC_HTTPBASIC_INTERNAL)
+			.users(
+					ALL_INDICES_TITLE_STARS_READER, TITLE_ARTIST_LYRICS_READER_USER, TITLE_READER,
+					ALL_INDICES_MASKED_TITLE_ARTIST_READER, MASKED_ARTIST_LYRICS_READER,
+					ALL_INDICES_STRING_ARTIST_READER, ALL_INDICES_STARS_LESS_THAN_ZERO_READER, TWINS_FIRST_ARTIST_READER
+			)
+			.build();
+
+	/**
+	* Function that returns id assigned to song with title equal to given title or throws {@link RuntimeException}
+	* when no song matches.
+	*/
+	static final BiFunction<Map<String, Song>, String, String> FIND_ID_OF_SONG_WITH_TITLE = (map, title) -> map.entrySet()
+			.stream().filter(entry -> title.equals(entry.getValue().getTitle()))
+			.findAny()
+			.map(Map.Entry::getKey)
+			.orElseThrow(() -> new RuntimeException("Cannot find id of song with title: " + title));
+
+	/**
+	* Function that returns id assigned to song with artist equal to given artist or throws {@link RuntimeException}
+	* when no song matches.
+	*/
+	static final BiFunction<Map<String, Song>, String, String> FIND_ID_OF_SONG_WITH_ARTIST = (map, artist) -> map.entrySet()
+			.stream().filter(entry -> artist.equals(entry.getValue().getArtist()))
+			.findAny()
+			.map(Map.Entry::getKey)
+			.orElseThrow(() -> new RuntimeException("Cannot find id of song with artist: " + artist));
+
+	static final TreeMap<String, Song> FIRST_INDEX_SONGS_BY_ID = new TreeMap<>() {{
+		put(FIRST_INDEX_ID_SONG_1, SONGS[0]);
+		put(FIRST_INDEX_ID_SONG_2, SONGS[1]);
+		put(FIRST_INDEX_ID_SONG_3, SONGS[2]);
+		put(FIRST_INDEX_ID_SONG_4, SONGS[3]);
+	}};
+
+	static final TreeMap<String, Song> SECOND_INDEX_SONGS_BY_ID = new TreeMap<>() {{
+		put(SECOND_INDEX_ID_SONG_1, SONGS[3]);
+		put(SECOND_INDEX_ID_SONG_2, SONGS[2]);
+		put(SECOND_INDEX_ID_SONG_3, SONGS[1]);
+		put(SECOND_INDEX_ID_SONG_4, SONGS[0]);
+	}};
+
+	@BeforeClass
+	public static void createTestData() {
```

Comment:
These tests are very complex to setup, and the tests themselves are verifying many different behaviors - its hard to judge the correctness of a test as written.

Can we isolate the FLS testing into different blocks, or instead orient them so they are more cleanly verified.  Diving into the exact request/response isn't needed, but making sure we the field is present or not is the test.  Using functions to abstract these stages of the test will make them approachable

```java
public void testFlsEnabledFieldsAreHiddenForNormalUsers() {
   // setup
   //   Create Index A, with Document B1, B2, with fields C1, C2, C3
   //    For the test user enforce FLS on C2
   // action & verify
   //   Search query plain - verify C2 isn't visible
   //   Search query aggregation - verify C2 isn't visible
   //   ...
}
```

Reply:
Done: https://github.com/opensearch-project/security/pull/2258/commits/8579eabed13d2722f2aaf83d8e89cce50ea54a4c